src/Algorithm.ts

```diff
@@ -25,11 +26,20 @@ export default class Algorithm extends Builder {
 
     let innerHTML = node.innerHTML; // TODO use original slice, forward this from linter
 
-    // prettier-ignore
-    const emdTree =
-      'ecmarkdownTree' in node
-        ? (node as any).ecmarkdownTree
-        : emd.parseAlgorithm(innerHTML, { trackPositions: true });
+    let emdTree;
+    if ('ecmarkdownTree' in node) {
+      emdTree = (node as any).ecmarkdownTree;
+    } else {
+      try {
+        emdTree = emd.parseAlgorithm(innerHTML);
+      } catch (e) {
+        warnEmdFailure(spec.warn, node, e);
+      }
+    }
+    if (emdTree == null) {
+      node.innerHTML = `#### ECMARKDOWN PARSE FAILED ###<pre>${innerHTML}</pre>`;
```

Comment:
```suggestion
      node.innerHTML = `### ECMARKDOWN PARSE FAILED ###<pre>${innerHTML}</pre>`;
```

Reply:
There's a couple cases of these. You might want to extract this to a constant and share it. Also, for the record, I believe the proper way to do this would be to add this content using CSS `:before { content: ... }` since it's entirely presentational.